### PR TITLE
[master < T1147 ] Improve NameToId mapper on set properties

### PR DESCRIPTION
**Pull request by antoniofilipovic**
This PR adds caching of name id mapper for property creation. If `name_id_mapper` proves to be a deadlock in other queries, we can think about how to add caching per storage accessor level. Thread local caching for `name_id_mapper` didn't yield so much better results.

Short benchmark.
The following queries were executed:
1. `FOREACH (i in range(0, 1000000) | CREATE (n:Label {id:i}));`
2. `CREATE INDEX ON :Label(id);`
3. `FOREACH (i IN range(0, 1000000 , 3) | MERGE (n:Label {id:i}) SET n += {prop2:"nmrtRnZQzm", prop3:"bielFtyXca", prop4:"AoCYiwqjuu", prop5:"GqyKUebYyi", prop6:"spdWPDdyiY", prop7:"dzpACFXOal", prop8:"NchfJxgMHL", prop9:"JKbOEGOhkl", prop10:"ftmaTyKESy", prop11:"cIgjUrbHcF"});`

Execution for last query:
- baseline: 
- this PR: 2.4 sec
- thread local caching (also this PR but two bugy reverted commits): 2.6 sec

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/query/plan/operator.cpp [hunk 1]
```diff
@@ -2808,9 +2809,15 @@ void SetPropertiesOnRecord(TRecordAccessor *record, const TypedValue &rhs, SetPr
...
-        auto key = context->db_accessor->NameToProperty(prop_id);
-        new_properties.emplace(key, prop_value);
+      for (const auto &[string_key, value] : rhs.ValueMap()) {
+        storage::PropertyId property_id;
+        if (auto it = cached_name_id.find(std::string(string_key)); it == cached_name_id.end()) {
+          property_id = context->db_accessor->NameToProperty(string_key);
+          cached_name_id.emplace(string_key, property_id);
+        } else {
+          property_id = it->second;
+        }
```
Comment (Josipmrden) [inline src/query/plan/operator.cpp:2819]: see if we need to reverse this if, depending on the frequency of hitting the cache vs. having to go to the storage

File: src/query/plan/operator.cpp [hunk 2]
```diff
@@ -2838,6 +2845,7 @@ bool SetProperties::SetPropertiesCursor::Pull(Frame &frame, ExecutionContext &co
   if (!input_cursor_->Pull(frame, context)) return false;
 
   TypedValue &lhs = frame[self_.input_symbol_];
+  // context.frame_change_collector->AddTrackingKey("a");
```
Comment (Josipmrden) [inline src/query/plan/operator.cpp:2848]: what's this

**Review by Josipmrden — APPROVED** [review]

**Comment by vpavicic** [comment]
@antoniofilipovic please can you give me a release note (RN)?

**Comment by antoniofilipovic** [comment]
@vpavicic yes, sorry. Performance on set properties is additionally improved by caching mappings from property name to internal property id

**Pull request merged**